src/matrix_content_scanner/scanner/scanner.py

```diff
@@ -309,6 +311,55 @@ async def _scan_file(
 
         return media
 
+    async def scan_file_on_disk(
+        self, file_path: str, metadata: Optional[JsonDict] = None
+    ) -> None:
+        """Scan a file that already exists on disk. The file will be deleted after scanning.
+
+        This does not cache the result.
+
+        Args:
+            file_path: The full file path to the source file.
+            metadata: The metadata attached to the file (e.g. decryption key), or None if
+                the file isn't encrypted.
+
+        Raises:
+            FileDirtyError if the result of the scan said that the file is dirty.
+        """
+        scan_filename = file_path
+        if metadata is not None:
+            with open(file_path, "rb") as f:
+                content = f.read()
+                # If the file is encrypted, we need to decrypt it before we can scan it.
+                media_content = self._decrypt_file(content, metadata)
+                scan_filename = self._write_file_to_disk(
+                    str(uuid.uuid4()), media_content
+                )
+
+                # Remove source file now we've decrypted it.
+                removal_command_parts = self._removal_command.split()
+                removal_command_parts.append(file_path)
+                subprocess.run(removal_command_parts)
```

Comment:
would be quite tempted to pull this out into its own function, to make it easier to modify the removal logic in future

(and if we're being pedantic, really this should be non-blocking/async, but I appreciate this is probably what was already here)